### PR TITLE
Reimplement using RepeatSensor

### PR DESCRIPTION
**Pull request by mairas**
Reimplement the add-on using RepeatSensor. Since the MAX31856 data acquisition is a two-stage process (first, launch the conversion, then come back after a short time to read the result), I think it was worth it doing a separate add-on library for it. However, the library can be implemented with RepeatSensor, making the implementation quite concise.

### CODE REVIEW

File: examples/max31856_example.cpp [hunk 1]
```diff
@@ -51,14 +55,19 @@ ReactESP app([]() {
   // Create a sensor that is the source of our data, that will be read every
   // 1000 ms.
   const uint readDelay = 1000;
```
Comment (ba58smith) [inline examples/max31856_example.cpp]: Should be read_delay to be consistent with the style sheet, right?

File: examples/max31856_example.cpp [hunk 2]
```diff
@@ -51,14 +55,19 @@ ReactESP app([]() {
...
+  // https://github.com/SignalK/SensESP/blob/master/src/sensors/max31856_thermocouple.h
+  Adafruit_MAX31856* ada_max31856 = new Adafruit_MAX31856(
+      SPI_CS_PIN, SPI_MOSI_PIN, SPI_MISO_PIN, SPI_CLK_PIN);
+
+  // thermocouple type:  MAX31856_TCTYPE_K;
+  // other types can be B, E, J, N, R, S, T
+  ada_max31856->setThermocoupleType(MAX31856_TCTYPE_K);
+
+  // Create the sensor object
+  auto* max31856 = new MAX31856(ada_max31856, readDelay);
```
Comment (ba58smith) [inline examples/max31856_example.cpp]: read_delay

**Review by ba58smith — APPROVED** [review]
The reimplementation is way above my pay grade, so I can't even guess if it's going to work. :-)

One style comment, otherwise LGTM.

File: examples/max31856_example.cpp (hunk 1 above)
Comment (mairas) [reply examples/max31856_example.cpp]: Sharp eyes! Missed that. Thanks.

**Pull request merged**